R/plotRD.r

```diff
@@ -24,115 +24,180 @@ plotRD <- function(distr, track, ptt, known=NULL, g, xlims, ylims, makePlot = TR
+      # get some plot indices
+      idx1 <- which((sv+1) == 1)
+      idx2 <- which((sv+1) == 2)
+      TT <- length(track$lon)
+      
+      if(save.plot) grDevices::pdf(paste(ptt, '_RD.pdf', sep = ''), width = 7, height = 12)
+      
+      # BUILD IT
+      nf <- layout(matrix(c(1,2,
+                            3,4,
+                            5,6), 3, 2, byrow=T), widths=c(7,2), heights=c(4,4,4))
+      #layout.show(nf)
+      
+      # COMBINED PANEL
+      par (mar=mar.bottom)
+      
+      raster::image(all, maxpixels=raster::ncell(all), xlim=xlims, ylim=ylims, axes=F,
+                    col=plot.rd.col, breaks=norm.breaks, xlab='', ylab='')#, main='All')#, zlim=c(.05,1))#, axes=F)#, main=dt.idx) #, breaks=zbreaks
+      axis(1, at=x.at, labels=FALSE)
+      axis(2, at=y.at, labels=y.labels);
+      world(add=T, fill=T, col='grey60')
+      box()
+      if(!is.null(known)) lines(known$lon, known$lat)
+      
+      # COLORBAR
+      image(1, norm.mid, t(as.matrix(norm.mid)), breaks=norm.breaks, col=plot.rd.col, axes=FALSE, xlab="",
+            ylab='Expected Proportion of Time Spent')
+      axis(2, at=c(.05, .25, .5, .75, .95), labels=paste(rev(c(5, 25, 50, 75, 95)), '%'));box();
+      
+      # MIGRATORY PANEL
+      par (mar=mar.bottom)
+      raster::image(norm[[1]], maxpixels=raster::ncell(norm[[1]]), xlim=xlims, ylim=ylims, axes=F,
+                    col=plot.rd.col, breaks=norm.breaks, xlab='', ylab='')#, main='Migratory')#, zlim=c(.05,1))#, axes=F)#, main=dt.idx) #, breaks=zbreaks
+      axis(1, at=x.at, labels=FALSE)
+      axis(2, at=y.at, labels=y.labels)
+      world(add=T, fill=T, col='grey60')
+      box()
+      mtext('Migratory behavior', 4, line=2)
+      graphics::points(track$lon[idx1], track$lat[idx1], bg='white', pch=21, cex=1.25)
+      graphics::points(track$lon[1], track$lat[1], bg = 'green', pch = 21, cex=1.75)
+      graphics::points(track$lon[TT], track$lat[TT], bg = 'red', pch = 21, cex=1.75)
+      
+      # FILL COLORBAR SPACE
+      plot.new()
+      
+      # RESIDENT PANEL
+      par (mar=mar.default)
+      raster::image(norm[[2]], maxpixels=raster::ncell(norm[[2]]), xlim=xlims, ylim=ylims, axes=F,
+                    col=plot.rd.col, breaks=norm.breaks, xlab='', ylab='')#, main='Resident')#, zlim=c(.05,1))#, axes=F)#, main=dt.idx) #, breaks=zbreaks
+      axis(1, at=x.at, labels=x.labels)
+      axis(2, at=y.at, labels=y.labels);
+      world(add=T, fill=T, col='grey60')
+      box()
+      mtext('Resident behavior', 4, line=2)
+      graphics::points(track$lon[idx2], track$lat[idx2], bg='grey60', pch=21, cex=1.25)
+      graphics::points(track$lon[1], track$lat[1], bg = 'green', pch = 21, cex=1.75)
+      graphics::points(track$lon[TT], track$lat[TT], bg = 'red', pch = 21, cex=1.75)
+      
+      # FILL COLORBAR SPACE
+      plot.new()
+      
+      if(save.plot) grDevices::dev.off()
+    }
     
-    # FILL COLORBAR SPACE
-    plot.new()
+    return(list(allRD = all, behavRD = norm))
```

Comment:
`return()` is typically on the second to last line right before the `}` that closes the function. Here, you're `return`ing inside the 2 state `else` statement which may cause weird things to happen. Second, we want the 1 state version to also `return` the same information. Thus, my vote would be to create both of these variables inside the `if else` then return them at the end. Obviously, in the 1 state version `behavRD = NULL`.

Reply:
Having navigated to my wd in terminal and run 
```
git remote -v
```
I see my origin listed as https://github.com/marosteg/HMMoce.git
and my upstream listed as https://github.com/camrinbraun/HMMoce.git

If I then attempt
```
git push
```
I get the error - fatal: The current branch state-plot-fixes has no upstream branch. Not sure why that error is flagging given that the origin and upstream are clearly assigned.

Reply:
hard to know without seeing what's happening on your end. what's the output of `git status`?

if you've already committed your additional changes to your *local* branch of `state-plot-fixes` then you shoudl just need:
```
git push -u origin state-plot-fixes
```

Reply:
'git status' yields:
```
On branch state-plot-fixes
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   .Rhistory

no changes added to commit (use "git add" and/or "git commit -a")
```
and 'git push -u origin state-plot-fixes' yields:
```
To https://github.com/marosteg/HMMoce.git
 ! [rejected]        state-plot-fixes -> state-plot-fixes (non-fast-forward)
error: failed to push some refs to 'https://github.com/marosteg/HMMoce.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Reply:
you need to `Integrate the remote changes` because `the tip of your current branch is behind its remote counterpart.`. I think you need `git fetch upstream`